src/main/java/controller/AttendanceController.java

```diff
@@ -0,0 +1,180 @@
+
+public class AttendanceController {
+    private static final String ATTENDANCE_HISTORY_FILE_NAME = "attendances.csv";
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final AttendanceSystem attendanceSystem;
+
+    public AttendanceController(final InputView inputView, final OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        final List<String> attendanceLines = FileManager.readFileLines(ATTENDANCE_HISTORY_FILE_NAME);
+        attendanceLines.remove(0);
+        attendanceSystem = AttendanceSystem.of(attendanceLines, LocalDate.now().withYear(2024).withMonth(12));
+    }
+
+    public void run() {
+        retryUntilOperationQuit();
+    }
+
+
+    public Operation selectOperation() {
+        outputView.printToday(LocalDate.now().withYear(2024).withMonth(12));
+        outputView.printIntroduceOperation();
+        final Operation operation = inputView.readChoiceOperation();
+        if (operation == Operation.ADD_ATTENDANCE) {
+            addAttendance();
+        } else if (operation == Operation.UPDATE_ATTENDANCE) {
+            updateAttendance();
+        } else if (operation == Operation.LOOKUP_CREW_ATTENDANCE) {
+            responseCrewAttendanceHistory();
+        } else if (operation == Operation.LOOKUP_EXPULSION_CREWS) {
+            responseExpulsionCrews();
+        }
+        return operation;
+    }
+
+    private void retryUntilOperationQuit() {
+        while (selectOperation() != Operation.QUIT) {
+        }
+    }
+
+
+    private void addAttendance() {
+        if (attendanceSystem.isNotAttendanceDay(LocalDate.now().withYear(2024).withMonth(12))) {
+            outputView.printNotAttendanceDay();
+            return;
+        }
+        final String crewName = LoopTemplate.tryCatchLoop(this::inputCrewName, outputView);
+        if (!attendanceSystem.isAlreadyTodayAttendance(crewName)) {
+            final Attendance attendance = LoopTemplate.tryCatchLoop(this::attendance, crewName, outputView);
+            final AttendanceResponse attendanceResponse = convertAttendanceToResponse(attendance);
+            outputView.printCrewAttendances(List.of(attendanceResponse));
+            return;
+        }
+        updateAttendanceForDuplicateAttendance(crewName);
+    }
+
+    private void updateAttendanceForDuplicateAttendance(final String crewName) {
+        outputView.printIntroduceAnswerCommand();
+        final AnswerCommand answerCommand = inputView.readAnswerCommand();
+        if (answerCommand == AnswerCommand.YES) {
+            final int dayOfMonth = LocalDate.now().getDayOfMonth();
+            final LocalTime targetTime = LoopTemplate.tryCatchLoop(this::inputUpdateTime, outputView);
+            final Attendance beforeAttendance = attendanceSystem.findAttendanceByDate(crewName, dayOfMonth);
+            final Attendance afterAttendance = attendanceSystem.updateAttendanceByCrewNameAndDay(targetTime, crewName,
```

Comment:
아래와 같이 찾아온 attendance 객체에게 출석 시간을 수정하라고 메시지를 보내보면 되지 않을까요? 😃 
```suggestion
            final Attendance attendance = attendanceSystem.findAttendanceByDate(crewName, dayOfMonth);
            attendance.updateAttenceTime(targetTime);
```

Reply:
> controller가 attendanceSystem에게 해당 크루의 출석을 변경시켜라 -> attendanceSystem은 크루를 찾아서 해당 날짜의 출석을 변경하라 -> 크루는 자신의 출석을 변경한다
위의 과정이 어색한걸까요?

find와 update가 두 번 이뤄진 것은, before와 after의 정보가 모두 필요했기 때문인데, 제가 잘못이해한걸까요?
로키의 생각이 궁금합니다.

Reply:
아하 ~  왜 부기가 이렇게 작성을 하신건지 이제 이해가 되네요. 

저는 after의 정보만 알면 된다고 생각했었는데, 요구사항에 따르면 before의 정보도 필요하기 때문이었겠군요? 😃

이 부분은 제가 제안드린 방식에서 `updateAttendanceTime` 의 반환 값을 전달해주면 될 것 같은데요. 
예를들자면 `AttendanceUpdateSnapshot` 같은 타입을 반환하여 필요한 before, after의 정보를 가지도록해볼 수도 있을 것 같은데, 이렇게한다면 저의 의도와 부기의 의도가 모두 만족되는게 맞을까요?

Reply:
말씀해주신 것 처럼 비즈니스 로직을 controller에서 도메인 객체를 제어해서 여러 단계로 수행하고 있습니다. 따라서 객체 내부에서 수행하는 것이 아닌 외부 controller에서 수행하고 있어 모호하긴 합니다. 
다만, 말씀하신 부분에서 before과 after를 묶은 `AttendanceUpdateSnapshot`을 만드신다고 하셨는데 이 경우 객체가 상태는 가지지만 행위는 가지지 않을 것 같습니다.
그럼 dto 처럼 데이터를 전달하기 위한 수단으로 사용하는데 도메인 객체가 dto를 만들어 반환하는게 어색한 것 같습니다.

이에 대한 로키의 생각이 궁금합니다! 🤔

Reply:
지금의 상황에서는 제가 말한대로라면 DTO 처럼 보일 수 있을 것 같아요.  😢
왜냐하면 적절히 위임해줄 역할이 당장 떠오르지 않기 때문에 그냥 단순히 필요한 데이터만 가진 객체가 되기 때문이죠. 
모든 객체가 역할과 책임을 가지면 객체지향적인 설계를 잘 한 것 같아 기분이 좋겠지만, 사실 모든 객체가 이상적으로 역할, 책임을 가지는게 생각보다 쉽지는 않다고 생각해요. 😅

하지만 지금처럼 before, after의 스냅샷을 Controller에서 각각 남겨서 비즈니스 로직을 수행하는 것 보다는 
 Attendance 객체가 직접 이 두 값을 기록해주는 것이 좋다고 생각해요. 🤔

Reply:
아하 dto처럼 보일 수 있지만, "스냅샷을 기록한다."라는 비즈니스 로직을 수행하기 위해 dto 처럼 보이는 객체를 기록해서 반환하는 것이군요 !!

>  사실 모든 객체가 이상적으로 역할, 책임을 가지는게 생각보다 쉽지는 않다고 생각해요. 😅

이것때문에 객체지향이 어려운 것 같아요.. 감사합니다!!